pinc/DifferenceEngineWrapperBB.inc

```diff
@@ -0,0 +1,29 @@
+<?php
+// This file inherits from DifferenceEngineWrapper to facilitate
+// formatting diffs using BB code.
+include_once($relPath."misc.inc"); // html_safe() in following includes
+include_once($relPath."prefs_options.inc"); // get_user_proofreading_font()
+include_once($relPath."DifferenceEngineWrapper.inc"); // get_user_proofreading_font()
+include_once("BBDiffFormatter.inc");
+
+class DifferenceEngineWrapperBB extends DifferenceEngineWrapper
+{
+    public function __construct($diffurl)
+    {
+        parent::__construct(new BBDiffFormatter($diffurl));
+    }
+
+    protected function addHeader($diff, $otitle, $ntitle, $multi = '', $notice = '')
+    {
+        $header = "";
+
+        if (!$diff && !$otitle) {
+            $header .= "{$ntitle}";
+        } else {
+            $header .= "
+            {$otitle}<br>
+            {$ntitle}";
```

Comment:
Can we make this and line 27 consistent? Either building the string all in one go and having implicit newlines or doing it piecemeal and being explicit about newlines -- either is fine but the mix is confusing. In general the code is explicit about newlines but we have a mix of building strings piecemeal or doing it all in one go, so whatever you want to do there is fine.

eg:
```php
$header .= "\n$otitle<br>\n$ntitle";
return "$header\n<pre>$diff</pre>\n";
// or
$header .= "\n{$otitle}<br>\n{$ntitle}";
return "{$header}\n<pre>{$diff}</pre>\n";
// or
$header .= "\n" . $otitle . "<br>\n" . $ntitle";
return $header . "\n<pre>" . $diff . "</pre>\n";
```